### ISSUE TITLE
[tune] Tried to create an owned object that already exists error

### ISSUE THREAD

**Issue by ejunprung** [ray-project/ray#9551]
Hi, I was curious to know what could cause this error. 

```
F0717 18:58:38.664489   457   457 reference_count.cc:155]  Check failed: object_id_refs_.count(object_id) == 0 Tried to create an owned object that already exists: 602bc3dce737ca0f7e0a4dfc010000c001000000
*** Check failure stack trace: ***
    @     0x7f948f3cd61d  google::LogMessage::Fail()
    @     0x7f948f3cea8c  google::LogMessage::SendToLog()
    @     0x7f948f3cd2f9  google::LogMessage::Flush()
    @     0x7f948f3cd511  google::LogMessage::~LogMessage()
    @     0x7f948f3aade9  ray::RayLog::~RayLog()
    @     0x7f948f161508  ray::ReferenceCounter::AddOwnedObject()
    @     0x7f948f14676b  ray::TaskManager::AddPendingTask()
    @     0x7f948f123f6e  ray::CoreWorker::SubmitActorTask()
    @     0x7f948f0bf0d2  __pyx_pw_3ray_7_raylet_10CoreWorker_43submit_actor_task()
    @     0x55850185b6e4  _PyMethodDef_RawFastCallKeywords
    @     0x55850185b86f  _PyMethodDescr_FastCallKeywords
    @     0x5585018b707c  _PyEval_EvalFrameDefault
    @     0x5585017f84f9  _PyEval_EvalCodeWithName
    @     0x55850185aa27  _PyFunction_FastCallKeywords
    @     0x5585018b38fe  _PyEval_EvalFrameDefault
    @     0x5585017f8db9  _PyEval_EvalCodeWithName
    @     0x55850185aa27  _PyFunction_FastCallKeywords
    @     0x5585018b2846  _PyEval_EvalFrameDefault
    @     0x5585017f8db9  _PyEval_EvalCodeWithName
    @     0x55850185aa27  _PyFunction_FastCallKeywords
    @     0x5585018b2ad0  _PyEval_EvalFrameDefault
    @     0x5585017f84f9  _PyEval_EvalCodeWithName
    @     0x55850185aa27  _PyFunction_FastCallKeywords
    @     0x5585018b2ad0  _PyEval_EvalFrameDefault
    @     0x55850185a79b  _PyFunction_FastCallKeywords
    @     0x5585018b2ad0  _PyEval_EvalFrameDefault
    @     0x55850185a79b  _PyFunction_FastCallKeywords
    @     0x5585018b2ad0  _PyEval_EvalFrameDefault
    @     0x55850185a79b  _PyFunction_FastCallKeywords
    @     0x5585018b2ad0  _PyEval_EvalFrameDefault
    @     0x55850185a79b  _PyFunction_FastCallKeywords
    @     0x5585018b2ad0  _PyEval_EvalFrameDefault
```

*Ray version and other system information (Python version, TensorFlow version, OS):*

Ray 0.8.6
Python 3.7.3
Tensorflow 1.15.0
Debian

**Comment by richardliaw** [comment]
@ejunprung can you provide a reproducible example (or a longer stack trace)?

also, can you maybe try the latest master?

**Comment by ejunprung** [comment]
@richardliaw Unfortunately, I'm not sure how to reproduce and the above is all I see in our logs. It happens maybe once every 100 times (I think) so it's kind of random. 

Could PBT checkpoints conflict with periodic checkpoints (checkpoint_freq) since we use both? I'll try the latest master but if there's anything obvious that I can investigate, I'd really appreciate it.

**Comment by richardliaw** [comment]
Yeah I haven't seen this before. It'd be great to see this on master; we've made a couple changes to the underlying ref-counting algorithm since the last release.

Can you tell me more about how this came about? i.e., what algo were you running, how long, etc

**Comment by ejunprung** [comment]
Ah, gotcha. We're using PPO and the error occurs maybe 3 - 4 hours into training. But it doesn't happen very often, it seems random. 

I'll keep an eye out and see if I can find any patterns. I'll close this for now. Thanks for the help!

**Comment by ejunprung** [comment]
Hi @richardliaw, quick question. When does the ref-counting algorithm come into play? I'm curious if there's anything in our tune config that could be causing this issue.

**Comment by ejunprung** [comment]
Hi @richardliaw, we've upgraded to Ray 1.0 but this error still occurs.

However, I think we've figured out the problem. Since we use AWS spot instances, training can abruptly stop and resume at any given moment. When this happens, I think it messes up Ray's reference counting. The solution is to either use dedicated AWS instances that don't cut out randomly or restart the training from scratch.

This is just an FYI in case this is something Ray could fix. We can deal with it on our end for now.
